controllers/machinehealthcheck_controller.go

```diff
@@ -151,7 +176,55 @@ func (r *MachineHealthCheckReconciler) reconcile(_ context.Context, cluster *clu
 		UID:        cluster.UID,
 	})
 
-	return ctrl.Result{}, fmt.Errorf("controller not yet implemented")
+	logger := r.Log.WithValues("machinehealthcheck", m.Name, "namespace", m.Namespace)
+	logger = logger.WithValues("cluster", cluster.Name)
+
+	// Create client for target cluster
+	clusterClient, err := remote.NewClusterClient(ctx, r.Client, cluster, r.scheme)
```

Comment:
From what I recall, kubelets post Node status updates every 10 seconds. That presumably will translate to this being called pretty frequently. We need to be aware of that fact, and make sure this isn't too expensive - just something to keep in mind.

Reply:
Interestingly, when I've tested this on a real cluster, I didn't see it reconciling too frequently. I saw it reconciling when the node conditions changed state (which is the desired effect) but it wasn't reconciling every 10s as you would expect IIRC.

I'll try spin up a test cluster again and double check this